packages/ui/src/helpers/authenticator/utils.ts

```diff
@@ -26,6 +27,24 @@ export const censorPhoneNumber = (val: string): string => {
   return split.join('');
 };
 
+export const censorContactInformation = (
+  type: ContactMethod,
+  value: string
+): string => {
+  let newVal = value;
+
+  if (type === 'Phone Number') {
+    newVal = censorPhoneNumber(value);
+  } else if (type === 'Email') {
+    const splitEmail = value.split('@');
+    const censoredName = censorAllButFirstAndLast(splitEmail[0]);
+
+    newVal = `${censoredName}@${splitEmail[1]}`;
+  }
+
+  return newVal;
+};
```

Comment:
IMO we should break this down in to different utils for slightly improved testability, removal of the extraneous mutability of the return value and update the naming of `censorContactInformation` to be more indicative of its use case, e.g.

```ts
export const censorEmail = (email: string): string => {
  const [username, domain] = email.split('@');
  return `${censorAllButFirstAndLast(username)}@${domain}`;
};

export const censorContactMethod = (
  type: ContactMethod,
  value: string
): string =>
  type === 'Phone Number' ? censorPhoneNumber(value) : censorEmail(value);
```